packages/driver/cypress/e2e/commands/assertions.cy.js

```diff
@@ -302,6 +302,19 @@ describe('src/cy/commands/assertions', () => {
         })
       })
 
+      // https://github.com/cypress-io/cypress/issues/22587
+      it('does not allow cypress commands inside the callback', (done) => {
+        cy.on('fail', (err) => {
+          expect(err.message).to.eq('`cy.should()` failed because you invoked a command inside the callback. Use `cy.then()` instead of `cy.should()`, or move any commands outside the callback function.\n\nThe command invoked was:\n\n  > `cy.log()`')
```

Comment:
Maybe we can educate the user as to why?

```suggestion
          expect(err.message).to.eq('`cy.should()` failed because you invoked a command inside the callback. `cy.should()` retries the inner function, but commands are already retried. Use `cy.then()` instead of `cy.should()`, or move any commands outside the callback function.\n\nThe command invoked was:\n\n  > `cy.log()`')
```

Reply:
Whoops, this only updated the test, not the error_messages. Added commit to address there as well (and also reworded slightly).